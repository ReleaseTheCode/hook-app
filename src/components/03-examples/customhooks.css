body {
    padding: 70px;    
}

body * {
    margin: 2px;
}