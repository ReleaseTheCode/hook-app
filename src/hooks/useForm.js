import { useState } from "react"

export const useForm = ( initialState ) => {
    const [values, setValues] = useState(initialState);

    const reset = () => {
        return setValues(initialState);
    }

    const handleInputChange = (evt) => {
        const { name, value } = evt.target;
        setValues( { ...values, [name]: value });
    }

    return [ values, handleInputChange, reset ];
}
