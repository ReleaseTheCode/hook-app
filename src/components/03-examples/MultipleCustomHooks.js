import React, { useEffect, useRef } from 'react'

import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import './customhooks.css'

export const MultipleCustomHooks = () => {
    
    const { counter, increment } = useCounter(1);
    const { data, loading } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
    
    const { quote, author } = ( !loading && !!data.length > 0 ) && data[0];
    
    return (
        <div>
            <h3>BreakingBad Quotes</h3>
            <hr/>
            { loading &&
                <div className='alert alert-info text-right'>
                    Loading...
                </div>
            }
            { !loading &&
                <>
                    <blockquote className='blockquote text-right'>
                        <p className='mb-2'>{ quote } </p>
                        <footer className='blockquote-footer'>{ author }</footer>
                    </blockquote>
                    <button 
                        className='btn btn-secondary'
                        onClick={increment}
                    >Next quote</button>
                </>
            }
        </div>
    )
}
