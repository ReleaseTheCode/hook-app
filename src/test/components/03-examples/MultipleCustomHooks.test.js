import React from 'react'
import { shallow } from 'enzyme'
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks'
import { useFetch } from '../../../hooks/useFetch';
jest.mock('../../../hooks/useFetch')

describe('Testing on MultipleCustomHooks', () => {
    test('should render correctly', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })
        const wrapper = shallow( <MultipleCustomHooks/> );
        expect(wrapper).toMatchSnapshot();
    });

    test('should render info', () => {
        useFetch.mockReturnValue({
            data: [{
                author: 'rich',
                quote: 'qwerty'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks/> );

        expect( wrapper.find('.alert').exists() ).toBe(false);
        expect( wrapper.find('.mb-2').text().trim() ).toBe('qwerty');

        console.log( wrapper.html() );
    });
    
});
