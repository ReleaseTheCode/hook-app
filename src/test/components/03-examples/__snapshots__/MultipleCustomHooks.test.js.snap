// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on MultipleCustomHooks should render correctly 1`] = `
<div>
  <h3>
    BreakingBad Quotes
  </h3>
  <hr />
  <div
    className="alert alert-info text-right"
  >
    Loading...
  </div>
</div>
`;
